pkg/webui/console/containers/device-importer/index.js

```diff
@@ -128,15 +168,43 @@ export default class DeviceImporter extends Component {
   }
 
   @bind
-  handleCreationProgress(device) {
+  handleCreationSuccess(device) {
     this.appendToLog(device)
-    this.setState(({ devicesComplete }) => ({ devicesComplete: devicesComplete + 1 }))
+    this.setState(({ devicesProcessed }) => ({
+      devicesProcessed: devicesProcessed + 1,
+    }))
   }
 
   @bind
-  handleError(error) {
-    const json = JSON.stringify(error, null, 2)
-    this.setState(({ log }) => ({ error, status: 'error', log: `${log}\n${json}` }))
+  handleCreationError(error) {
+    this.logError(error)
+    const { convertedDevices, devicesProcessed } = this.state
+    const currentDevice =
+      convertedDevices.indexOf(devicesProcessed) !== -1 ? convertedDevices[devicesProcessed] : {}
```

Comment:
What exactly is happening here? `convertedDevices` is an array of converted devices, while `devicesProcessed` is just a number. As far as I can tell, `currentDevice` is always an empty object because `indexOf` returns -1. Am I missing something?

Reply:
Yeah. `indexOf()` is not actually the right method here. A simple check against the length of the array should suffice. What this does is map the error to the corresponding device ID, using the index of the currently created device. This is necessary since the error itself does not contain information about what exact device ID it is connected to.

I'll fix and clarify this section.